src/components/Quiz.js

```diff
@@ -18,19 +18,48 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
+    let checkedAnswer;
+    const answerId = Number(e.target.value);
+    const previousSelectedAnswers = answers[e.target.name] || [];
+    const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
+      answerId,
+    );
+
+    if (isAnswerPreviouslySelected) {
+      const removePreviousAnswer = previousSelectedAnswers.filter(
+        answer => answer !== answerId,
+      );
+      checkedAnswer = {
+        [e.target.name]: removePreviousAnswer,
+      };
+    } else {
+      checkedAnswer = {
+        [e.target.name]: [...previousSelectedAnswers, answerId],
+      };
+    }
+    return checkedAnswer;
```

Comment:
We can simplify this even further! I've laid out the steps below since we could go back and forth a lot otherwise.

1. Remove `removePreviousAnswer` variable
```javascript
    let checkedAnswer;
    const answerId = Number(e.target.value);
    const previousSelectedAnswers = answers[e.target.name] || [];
    const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
      answerId,
    );

    if (isAnswerPreviouslySelected) {
      checkedAnswer = {
        [e.target.name]: previousSelectedAnswers.filter(answer => answer !== answerId),
      };
    } else {
      checkedAnswer = {
        [e.target.name]: [...previousSelectedAnswers, answerId],
      };
    }
    return checkedAnswer;
```

2. Move the object literal from `checkedAnswer` to `return`
```javascript
    let checkedAnswer;
    const answerId = Number(e.target.value);
    const previousSelectedAnswers = answers[e.target.name] || [];
    const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
      answerId,
    );

    if (isAnswerPreviouslySelected) {
      checkedAnswer = previousSelectedAnswers.filter(answer => answer !== answerId);
    } else {
      checkedAnswer = [...previousSelectedAnswers, answerId];
    }
    return {
      [e.target.name]: checkedAnswer,
    };
```

3. Turn the `if` statement to a ternary
```javascript
    const answerId = Number(e.target.value);
    const previousSelectedAnswers = answers[e.target.name] || [];
    const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
      answerId,
    );

    const checkedAnswer = isAnswerPreviouslySelected
      ? previousSelectedAnswers.filter(answer => answer !== answerId)
      : [...previousSelectedAnswers, answerId];

    return {
      [e.target.name]: checkedAnswer,
    };
```

4. Extract `e.target.name` to a variable since the name `e.target.name` isn't clear what it is
```javascript
    const questionId = e.target.name;
    const answerId = Number(e.target.value);
    const previousSelectedAnswers = answers[questionId] || [];
    const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
      answerId,
    );

    const checkedAnswer = isAnswerPreviouslySelected
      ? previousSelectedAnswers.filter(answer => answer !== answerId)
      : [...previousSelectedAnswers, answerId];

    return {
      [questionId]: checkedAnswer,
    };
```

5. Remove the `isAnswerPreviouslySelected` variable
```javascript
    const questionId = e.target.name;
    const answerId = Number(e.target.value);
    const previousSelectedAnswers = answers[questionId] || [];

    const checkedAnswer = previousSelectedAnswers.includes(answerId)
      ? previousSelectedAnswers.filter(answer => answer !== answerId)
      : [...previousSelectedAnswers, answerId];

    return {
      [questionId]: checkedAnswer,
    };
```

6. (Optional) Rename `checkedAnswer` and use `concat` instead
```javascript
    const questionId = e.target.name;
    const answerId = Number(e.target.value);
    const previousSelectedAnswers = answers[questionId] || [];

    const newCheckboxAnswer = previousSelectedAnswers.includes(answerId)
      ? previousSelectedAnswers.filter(answer => answer !== answerId)
      : previousSelectedAnswers.concat(answerId);

    return {
      [questionId]: newCheckboxAnswer,
    };
```

Reply:
/cc @AightA @kosenchiha — this might be interesting to you to see the incremental steps that happen when refactoring code

Reply:
Done!